hass_nabucasa/ice_servers.py

```diff
@@ -0,0 +1,140 @@
+"""Manage ICE servers."""
+
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+import logging
+import time
+from typing import TYPE_CHECKING
+
+from aiohttp import ClientResponseError
+from aiohttp.hdrs import AUTHORIZATION, USER_AGENT
+
+if TYPE_CHECKING:
+    from . import Cloud, _ClientT
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class IceServer:
+    """ICE Server."""
+
+    urls: str
+    username: str
+    credential: str
+
+
+class IceServers:
+    """Class to manage ICE servers."""
+
+    def __init__(self, cloud: Cloud[_ClientT]) -> None:
+        """Initialize ICE Servers."""
+        self.cloud = cloud
+        self._refresh_task: asyncio.Task | None = None
+        self._ice_servers: list[IceServer] = []
+        self._ice_servers_listener: Callable[[], Awaitable[None]] | None = None
+        self._ice_servers_listener_unregister: list[Callable[[], None]] = []
+
+        cloud.iot.register_on_connect(self.on_connect)
+        cloud.iot.register_on_disconnect(self.on_disconnect)
+
+    async def _async_fetch_ice_servers(self) -> None:
+        """Fetch ICE servers."""
+        if TYPE_CHECKING:
+            assert self.cloud.id_token is not None
+
+        async with self.cloud.websession.get(
+            f"https://{self.cloud.servicehandlers_server}/webrtc/ice_servers",
+            headers={
+                AUTHORIZATION: self.cloud.id_token,
+                USER_AGENT: self.cloud.client.client_name,
+            },
+        ) as resp:
+            if resp.status >= 400:
+                _LOGGER.error("Failed to fetch ICE servers: %s", resp.status)
```

Comment:
Why do we both log and raise below? I'd just raise and let users of this method decide how to log.

Reply:
Good point 👍🏻. We already log if this is raised.